docs/rfcs/0002-layer-architecture.md

```diff
@@ -0,0 +1,328 @@
+| **Hypervisor operating system**    | Run VM workloads, enable kernel-level security | [Defense in depth], [De-facto standards], [Declarative management], [Raspberry Pi compatibility], [Portability] |
+| **[CNI] compliant networking**     | Network the Racklet cluster compute nodes      | [Security by design], [No old/insecure protocols], [Openness], [Observability], [End-to-end encryption]         |
+| **GitOps tooling**                 | Declarative management of the Racklet stack    | [Improve status quo], [De-facto standards], [Declarative management], [Observability], [Auto-upgradability]     |
+
+[Kubernetes]: https://kubernetes.io/
+[CNI]: https://github.com/containernetworking/cni
+
+### 3. Firmware layer
+
+**Summary**: The firmware helps in securely booting and configuring Racklet compute, for example it is declaratively managed and performs cryptographic verification of payloads to boot. The firmware should also help with collecting hardware observability data and telemetry for monitoring and debugging.
+
+**Goals**:
+
+- Enable secure access to the declarative configuration in Git
+- Verify payloads to be booted by the compute
+- Enable debugging and observability of the hardware and compute
+- Store keys and signatures for the above layers
+
+**Layer components**:
+
+| Component                                | Role                                                                         | Key Requirements                                                                                                        |
+| ---------------------------------------- | ---------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------- |
+| **[u-root] based bootstrap environment** | Secure Git access, firmware updates and payload booting                      | [Security by design], [Improve status quo], [Open source], [Secure updates], [Zero-trust network boot]                  |
+| **BMC firmware**                         | Compute booting and debugging, key and signature storage for software layers | [Security by design], [No old/insecure protocols], [Declarative management], [Debuggability], [One-time hardware setup] |
+| **RMC firmware**                         | Rack hardware control and observability, e.g. fans                           | [Openness], [Declarative management], [Loose coupling], [Observability], [Secure updates]                               |
+
+*[BMC]: Baseboard Management Controller
+*[RMC]: Rack Management Controller
+
+[u-root]: https://u-root.org/
+
+### 2. Electrical layer
+
+**Summary**: The electrical layer backs the power delivery and physical networking requirements of the compute and provides a means to run the firmware on the BMC and RMC (microcontrollers).
```

Comment:
maybe spell out RMC and BMC at some point?

Reply:
Mouse over them :)